subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -845,13 +872,25 @@ static int start_job(struct nrf_cloud_fota_job *const job)
 		LOG_ERR("Failed to start FOTA download: %d", ret);
 		job->status = NRF_CLOUD_FOTA_FAILED;
 		job->error = NRF_CLOUD_FOTA_ERROR_DOWNLOAD_START;
-		send_event(NRF_CLOUD_FOTA_EVT_ERROR, job);
+		if (send_evt) {
+			send_event(NRF_CLOUD_FOTA_EVT_ERROR, job);
+		}
```

Comment:
I can imagine a customer might want to handle errors in the event-based manner even if using `NRF_CLOUD_FOTA_AUTO_START_JOB=n`

Generally I tend to personally feel that events should not try to take into account whether they are actually necessary for a given configuration case -- they should just fire when their described condition is met, and leave it up to the user to determine which of them are useful in a given use-case

But it's not a hard-line issue for me, I just notice an opportunity here to not introduce additional conditional behavior

Reply:
i did consider just using the events, but i opted for this method since it seemed simpler to directly handle the return value of `nrf_cloud_fota_job_start` function instead of getting the status via the event callback.

```
 * @retval 0            FOTA update job started successfully.
 * @retval -ENOTSUP     Error; @kconfig(CONFIG_NRF_CLOUD_FOTA_AUTO_START_JOB is enabled).
 * @retval -ENOENT      Error; FOTA update job information has not been received.
 * @retval -EINPROGRESS Error; a FOTA update job is in progress.
 * @return A negative value indicates an error starting the job.
```
i could add an explicit return error code for the NRF_CLOUD_FOTA_ERROR_DOWNLOAD_START case?

Reply:
Oh, to be clear, I'm totally fine with `int nrf_cloud_fota_job_start(void)` returning an error code and with `handle_mqtt_evt_publish` using that error code directly instead of events

I'm just suggesting essentially to remove the `send_evt` guard on the `send_event(NRF_CLOUD_FOTA_EVT_ERROR, job);` call -- basically fire it even if the user is directly calling `nrf_cloud_fota_job_start`